assets/app/scripts/controllers/builds.js

```diff
@@ -58,6 +58,72 @@ angular.module('openshiftConsole')
       }      
     };
 
+    // Function which will 'instantiate' new build from given buildConfigName
+    $scope.startBuild = function(buildConfigName) {
+      var req = {metadata:{name:buildConfigName}};
+      DataService.create("buildConfigs/instantiate", buildConfigName, req, $scope).then(
+        function(build) { //success
+            $scope.alerts = [
+            {
+              type: "success",
+              message: "Build " + build.metadata.name + " has started.",
+            }
+          ];
+        },
+        function(result) { //failure
+          $scope.alerts = [
+            {
+              type: "error",
+              message: "An error occurred while starting the build.",
+              details: "Status: " + result.status + ". " + result.data,
+            }
+          ];
+        }
+      );
+    };
+
+    // Function which will 'clone' build from given buildName
+    $scope.cloneBuild = function(buildName) {
+      var req = {metadata:{name:buildName}};
+      DataService.create("builds/clone", buildName, req, $scope).then(
+        function(build) { //success
+            $scope.alerts = [
+            {
+              type: "success",
+              message: "Build " + build.metadata.name + " has been rerun from build " + buildName + ".",
+            }
+          ];
+        },
+        function(result) { //failure
+          $scope.alerts = [
+            {
+              type: "error",
+              message: "An error occurred while rerunning the build.",
+              details: "Status: " + result.status + ". " + result.data,
+            }
+          ];
+        }
+      );
+    };
+
+    // Function to disable Start/Rebuild build button, which checks,
+    // if any other build from given buildConfig is running.
+    // Returns true/false
+    $scope.isBuildRunning = function(buildConfigName) {
```

Comment:
I worry about how often this is going to be getting called because of how often the angular digest loops will trigger and recheck this. And for every single build its rechecking if that buildConfig has a build running by checking the builds for the buildConfig.

What I would rather see is in the controller when the callback for buildConfigs or builds gets fired (any time we get new data from the server) update a map of $scope.buildConfigBuildInProgress or something to that effect, which is a map of buildConfig name to true/false

To make it even better, a watch callback can accept optional additional parameters for "action" and "object".  Where action is one of ADDED/DELETED/MODIFIED and object is the resource for that action.  Initially the callback will just be sent a list of data back and action and object will be null, and in this case you can build the buildConfigBuildsInProgress map. On subsequent calls to the callback when action/object are not null you know the specific item that got updated and can just update the relevant part of buildConfigBuildsInProgress.  You'd want to make updates any time either the buildConfig callback gets fired or the builds callback gets fired.